tests/components/iaqualink/test_init.py

```diff
@@ -0,0 +1,335 @@
+"""Tests for iAqualink integration."""
+
+import asyncio
+from unittest.mock import patch
+
+from iaqualink.device import (
+    AqualinkAuxToggle,
+    AqualinkBinarySensor,
+    AqualinkDevice,
+    AqualinkLightToggle,
+    AqualinkSensor,
+    AqualinkThermostat,
+)
+from iaqualink.exception import AqualinkServiceException
+
+from homeassistant.components.binary_sensor import DOMAIN as BINARY_SENSOR_DOMAIN
+from homeassistant.components.climate import DOMAIN as CLIMATE_DOMAIN
+from homeassistant.components.iaqualink import DOMAIN, UPDATE_INTERVAL, AqualinkEntity
+from homeassistant.components.light import DOMAIN as LIGHT_DOMAIN
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.util import dt as dt_util
+
+from tests.common import async_fire_time_changed
+from tests.components.iaqualink.conftest import (
+    async_raises,
+    async_returns,
+    get_aqualink_client,
+    get_aqualink_device,
+    get_aqualink_system,
+)
+
+
+def _ffwd_next_update_interval(hass):
+    now = dt_util.utcnow()
+    async_fire_time_changed(hass, now + UPDATE_INTERVAL)
+
+
+async def test_setup_login_exception(hass, config_entry):
+    """Test setup encountering a login exception."""
+    config_entry.add_to_hass(hass)
+
+    with patch(
+        "iaqualink.client.AqualinkClient.login",
```

Comment:
Please use a more exact patch target so we know we are patching only the place we are testing.

untested, but something like this should work
```suggestion
        "homeassistant.components.iaqualink.config_flow.AqualinkClient.login",
```